### PR TITLE
fix(bindings): base64 payload and key for content topic

### PR DESCRIPTION
**Pull request by richard-ramos**
# Description
Various fixes to the bindings to issues identified while attempting to integrate nwaku with [waku-rust-bindings](https://github.com/waku-org/waku-rust-bindings/pull/87)

# Changes
- [ ] Stores the userData that's passed when `waku_set_event_callback` is invoked
- [ ] Adds the messageHash to the response when an event is generated
- [ ] Converts from/to base64 the message payloads
- [ ] Uses the key `contentTopic` instead of `content_topic` in the message JSON

### CODE REVIEW

File: library/events/json_message_event.nim [hunk 1]
```diff
@@ -1,16 +1,59 @@
...
+    version: uint32(jsonContent["version"].getInt()),
+    timestamp: int64(jsonContent["timestamp"].getBiggestInt()),
+    ephemeral: jsonContent["ephemeral"].getBool(),
+    meta: Base64String(jsonContent["meta"].getStr()),
+  )
+
+proc toWakuMessage*(self: JsonMessage): WakuMessage =
+  let payloadRes = base64.decode(self.payload)
+  if payloadRes.isErr():
+    raise newException(ValueError, "invalid payload format: " & payloadRes.error)
```
Comment (SionoiS) [inline library/events/json_message_event.nim:37]:
```suggestion
  let payload = base64.decode(self.payload).valueOr:
    raise newException(ValueError, "invalid payload format: " & error)
```

File: library/events/json_message_event.nim [hunk 2]
```diff
@@ -1,16 +1,59 @@
...
+
+  let metaRes = base64.decode(self.meta)
+  if metaRes.isErr():
+    raise newException(ValueError, "invalid meta format: " & metaRes.error)
```
Comment (SionoiS) [inline library/events/json_message_event.nim:41]:
```suggestion
  let meta = base64.decode(self.meta).valueOr:
    raise newException(ValueError, "invalid meta format: " & error)
```

**Review by SionoiS — APPROVED** [review]
LGTM thanks!

File: library/events/json_message_event.nim (hunk 1 above)
Comment (Ivansete-status) [inline library/events/json_message_event.nim:37]: Nitpick, if possible I'd avoid using exceptions:)

File: library/events/json_message_event.nim [hunk 3]
```diff
@@ -1,16 +1,59 @@
...
+
+  WakuMessage(
+    payload: payloadRes.value,
+    meta: metaRes.value,
+    contentTopic: self.contentTopic,
+    version: uint32(self.version),
+    timestamp: self.timestamp,
+    ephemeral: self.ephemeral
+  )
```
Comment (Ivansete-status) [inline library/events/json_message_event.nim:50]: At first sight I missed having the rln proof but I believe this is something we will add in the future when we expose some kind of `publish_rln` function

File: library/libwaku.nim [hunk 4]
```diff
@@ -43,14 +45,14 @@ const RET_MISSING_CALLBACK: cint = 2
 proc relayEventCallback(ctx: ptr Context): WakuRelayHandler =
   return proc (pubsubTopic: PubsubTopic, msg: WakuMessage): Future[system.void]{.async.} =
     # Callback that hadles the Waku Relay events. i.e. messages or errors.
-    if not isNil(ctx[].eventCallback):
+    if not isNil(ctx[].eventCallback) and not isNil(ctx[].eventUserData):
```
Comment (Ivansete-status) [inline library/libwaku.nim:48]: Given that we now have two conditions that should be met, I wonder if we could split the validation in two, and apply some kind of  "return early pattern". The following article explains the idea much better:
https://medium.com/swlh/return-early-pattern-3d18a41bba8

lmk what do you think :)

**Review by Ivansete-status — APPROVED** [review]
LGTM! Thanks for it!
Just added a couple of nitpick comments xD

File: library/events/json_message_event.nim (hunk 1 above)
Comment (richard-ramos) [reply library/events/json_message_event.nim]: Fixed in https://github.com/waku-org/nwaku/pull/2435/commits/a4f7b88bacdbd30fe7a76654d4e3a732ddeb1f66

File: library/events/json_message_event.nim (hunk 2 above)
Comment (richard-ramos) [reply library/events/json_message_event.nim]: Fixed in https://github.com/waku-org/nwaku/pull/2435/commits/a4f7b88bacdbd30fe7a76654d4e3a732ddeb1f66

File: library/events/json_message_event.nim (hunk 1 above)
Comment (richard-ramos) [reply library/events/json_message_event.nim]: Replaced by a Result in https://github.com/waku-org/nwaku/pull/2435/commits/a4f7b88bacdbd30fe7a76654d4e3a732ddeb1f66

File: library/events/json_message_event.nim (hunk 3 above)
Comment (richard-ramos) [reply library/events/json_message_event.nim]: Ended up adding the proof in https://github.com/waku-org/nwaku/pull/2435/commits/a4f7b88bacdbd30fe7a76654d4e3a732ddeb1f66 so it's now 'future-ready' for when we include rln functionality to bindings

File: library/libwaku.nim (hunk 4 above)
Comment (richard-ramos) [reply library/libwaku.nim]: Sounds good! I removed the validation from the `if` and executed those following the 'return early' pattern as described in the article you shared.
https://github.com/waku-org/nwaku/pull/2435/commits/a4f7b88bacdbd30fe7a76654d4e3a732ddeb1f66

**Pull request merged**